hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/BaseSparkInternalRowReaderContext.java

```diff
@@ -170,6 +171,9 @@ public Comparable convertValueToEngineType(Comparable value) {
 
   @Override
   public InternalRow getDeleteRow(InternalRow record, String recordKey) {
-    throw new UnsupportedOperationException("Not supported for " + this.getClass().getSimpleName());
+    if (record != null) {
+      return record;
+    }
+    return new HoodieInternalRow(null, null, UTF8String.fromString(recordKey), null, null, null, false);
```

Comment:
we should also be able to pass in partition path as an arg and set it in the record right. 
Or the partition meta field is pretty much never used only(within file group reader) w/ all other recent changes we landed ?

Reply:
You're right, the partition path is present in the reader context so I have updated this line.